docs/tildagon-apps/widgets-and-hardware/ui-elements.md

```diff
@@ -0,0 +1,412 @@
+# UI Widget Overview
+
+You can use the following premade [`app_components`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/app_components/menu.py) to create user interfaces:
+
+## Menu
+
+The [`Menu`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/app_components/menu.py) component, allows you to create menus with which you can scroll through a list of options and select an item.
+
+### Example
+
+```python
+from app import App
+from app_components import Menu, Notification, clear_background
+
+main_menu_items = ["menu_item1", "menu_item2", "menu_item3"]
+
+class MenuDemo(App):
+    def __init__(self):
+        self.menu = Menu(
+            self,
+            main_menu_items,
+            select_handler=self.select_handler,
+            back_handler=self.back_handler,
+        )
+        self.notification = None
+
+    def select_handler(self, item):
+        self.notification = Notification('You selected "' + item + '"!')
+
+    def back_handler(self):
+        self.minimise()
+
+    def draw(self, ctx):
+        clear_background(ctx)
+        self.menu.draw(ctx)
+        if self.notification:
+            self.notification.draw(ctx)
+
+    def update(self, delta):
+        self.menu.update(delta)
+        if self.notification:
+            self.notification.update(delta)
+```
+
+You can see a more comprehensive example in the [`menu_demo.py`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/firmware_apps/menu_demo.py).
+
+
+### Usage
+
+To use a menu:
+
+1. Import the `Menu` component:
+
+    ```python
+    from app_components import Menu
+    ```
+
+2. Define your menu items above your App class:
+
+    ```python
+    menu_items = [ "menu_item1", "menu_item2" ]
+    ```
+
+3. Initialize a variable to hold the menu in the `__init__` method of your app:
+
+    ```python
+    self.menu = Menu(
+        self,
+        menu_items,
+        select_handler=self.select_handler,
+        back_handler=self.back_handler,
+    )
+    ```
+
+    To initialize the Menu use the following parameters:
+
+    | Parameter | Type | Description |
+    | --------- | ---- | ----------- |
+    | `app` | `App` | The app to add the menu to. |
+    | `menu_items` | `[str]` | A list of strings containing the menu items. |
+    | `select_handler` | `method` | The method to be called when an item is selected and the confirm button is pressed. |
+    | `back_handler` | `method` | The method to be called when the cancel button is pressed. |
+    | `position` | `int` | _Optional_. The menu position to start at. Default: `0`. |
+    | `speed_ms` | `int` | _Optional_. The speed to redraw the UI at. Default: `300`. |
+    | `item_font_size` | `float` | _Optional_. The item line height. Default: `26.04`. |
```

Comment:
26.04 seems like a random number, but it's a pt value: https://github.com/emfcamp/badge-2024-software/blob/main/modules/app_components/tokens.py#L9  - I wonder if we should tell people that while the real unit here is (I'm not sure because it's not documented in the [ctx docs](https://ctx.graphics/protocol/) but I think it's the height of the character in px), we have those utility sizes that they could use, or they could do 10*one_pt to get specific pt sizes?

Reply:
Yeah I was considering that, will add that to my to do list